assets/js/modules/analytics/datastore/settings.js

```diff
@@ -0,0 +1,181 @@
+/**
+ * modules/analytics data store: settings.
+ *
+ * Site Kit by Google, Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Internal dependencies
+ */
+import Data from 'googlesitekit-data';
+import {
+	isValidAccountID,
+	isValidInternalWebPropertyID,
+	isValidPropertySelection,
+	isValidProfileSelection,
+	isValidPropertyID,
+} from '../util';
+import { STORE_NAME, PROPERTY_CREATE, PROFILE_CREATE } from './constants';
+
+const { createRegistrySelector, createRegistryControl } = Data;
+
+// Actions
+const SUBMIT_CHANGES = 'SUBMIT_CHANGES';
+const START_SUBMIT_CHANGES = 'START_SUBMIT_CHANGES';
+const FINISH_SUBMIT_CHANGES = 'FINISH_SUBMIT_CHANGES';
+
+export const INITIAL_STATE = {
+	isDoingSubmitChanges: false,
+};
+
+export const actions = {
+	*submitChanges() {
+		yield {
+			payload: {},
+			type: START_SUBMIT_CHANGES,
+		};
+
+		yield {
+			payload: {},
+			type: SUBMIT_CHANGES,
+		};
+
+		yield {
+			payload: {},
+			type: FINISH_SUBMIT_CHANGES,
+		};
+	},
+};
+
+export const controls = {
+	[ SUBMIT_CHANGES ]: createRegistryControl( ( registry ) => async () => {
```

Comment:
Let's add a call to `api.invalidateCache( 'modules', 'analytics' )` to this function to ensure it invalidates any existing caches for when settings (e.g. account or property) changes. Otherwise API calls that rely on cached data (right now there are none, but we will have them for all insights) would serve data for potentially the wrong values.

We could be precise about this and only invalidate when the account, property, or profile changes, but I think it's most straightforward to just do it any time _any_ setting has changed.

Reply:
✅